lib/common/client/middleware/response/ppms_parser.rb

```diff
@@ -13,10 +13,16 @@ def on_complete(env)
 
           def parse_body(env)
             hash = JSON.parse(env.body)
-            # flag Not Found errors as success so the error doesn't bubble up
-            if hash['error'] && hash['error']['message'] && hash['error']['message'] =~ /No Providers found/
+            msg = hash.dig('error', 'message')
+
+            case msg
+            when /No Providers found/ # flag Not Found errors as success so the error doesn't bubble up
               env[:status] = 200
               return []
+            when /An error has occurred/ # PPMS has encountered an internal error
+              hash['error']['code'] = nil if hash['error']['code'].blank? # Set code so it can match exceptions.en.yml
+              hash['error']['detail'] = hash['error']['message']
+              return hash['error']
             end
```

Comment:
if msg doesn't match either case isn't [this error](http://sentry.vetsgov-internal/vets-gov/platform-api-production/issues/59689/) still going to occur where `body` is `nil`?

Reply:
I agree that we need to account for a potential issue where there is no body (however unlikely it is... since even in this case there WAS a body but it was just a stringified hash error message) that we return, at the very least, an empty hash so it doesnt fail on trying to index in the `raise_error` file.

Reply:
Yeah - if we're getting an error with an empty body that means PPMS is sending unexpected payloads. Would have to cross reference their API docs to cover any additional error cases, I imagine.

Reply:
@lihanli @SMLuthi I think it'd be more appropriate to handle the "empty body" case in a separate body of work, as it relates to other services/controllers. As discussed in the eng. mtg. today w/ @saneshark brought up better handling of "non-coded" errors, this would fall into that category.

Which, I think, would be best handled in a response middleware with accompanying shared_context examples for any 3rd party service class

Reply:
I think in the case of the empty / nil body, just making sure that the 500 response received can map to an appropriate response that can be properly handled is maybe satisfactory. A 500 error might be a 502 or 503. I'm certain we're handling this sort of thing elsewhere (presumably in Rx and SM APIs since those are most mature at this point), could take a look there and make sure we're handling things consistently?

Reply:
Yeah, like we have the VA900 code as a catch-all [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/config/locales/exceptions.en.yml#L174)

Reply:
@saneshark to be clear, this PR will make it such that `500` responses will raise a `BackendServiceException` with the `PPMS_500` key to identify. If a different `50x` status comes back we would see a `VA900` code which should prompt us to introduce further specific error handling for this endpoint.

This will help us, for the time being, to identify more precisely how often PPMS is returning this error.

--- 

In a future work effort we will investigate solutions to further streamline nil response checking, i.e. in a middleware layer.

We do have [`gids_errors`](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/common/client/middleware/response/gids_errors.rb) that has a similar pattern, though the errors body there is pluralized.

I think there may be a way to extract the task of checking for JSON errors into its own `json_errors` middleware, but given the inconsistencies between providers, it may be cleaner to follow a `{service}_errors.rb` middleware pattern as we're doing already with [`mhv_errors.rb`](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/common/client/middleware/response/mhv_errors.rb).